100-level-AsyncAPI-Intro/step2.md

```diff
@@ -0,0 +1,244 @@
+
+
+```yml
+components:
+  messages:
+    ChatMessage:
+      name: ChatMessage
+      title: Chat Message
+      contentType: application/json
+      payload:
+        $ref: '#/components/schemas/ChatMessagePayload'
+```
+
+- **Schemas (Structure of message content):**
+
+`Schemas` in AsyncAPI are like the blueprints for the spaceship. They define the structure of the message content, ensuring that everything fits together correctly. Each schema is associated with a component and describes the structure and content of the messages.
+
+```yml
+components:
+  schemas:
+    ChatMessagePayload:
+      type: object
+      properties:
+        userId:
+          type: string
+          description: The ID of the user who sent the message
+        message:
+          type: string
+          description: The chat message
+```
+
+---
+
+Certainly! Here's the updated hands-on exercise, guiding you through the entire process of creating and configuring Chan's AsyncAPI document for communication with Capuccinova and beyond.
+
+### Chan's Galactic Coding Adventure: Crafting the Capuccinova Connection
+
+#### Prologue: A New Beginning in the Stars
+
+Chan embarks on a mission to establish a sophisticated communication system with Eve on Capuccinova. With your expertise in AsyncAPI, you'll assist him in creating a robust AsyncAPI document from scratch.
+
+---
+
+#### Chapter 1: Initiating the Capuccinova Communication Link
+
+**Your Mission: Creating the Base AsyncAPI Document**
+
+1. **Start the Journey**: In the KillerCoda terminal, create a new file by typing `nano chan-to-capuccinova.yaml`.
+
+2. **Lay the Cosmic Foundations**: Enter the basic AsyncAPI structure:
+   ```yaml
+   asyncapi: 3.0.0
+   info:
+     title: Space Communication API
+     version: '1.0.0'
+     description: API for communication between Chan's spaceship and Capuccinova
+
+   servers:
+     production:
+       host: space.mosquitto.org
+       protocol: mqtt
+       description: Production server for space communication
+
+   channels:
+     roomidchat:
+       address: chat/{roomId}
+       messages:
+         chatMessage:
+           name: ChatMessage
+           payload:
+             type: object
+             properties:
+               userId:
+                 type: string
+                 description: The ID of the user who sent the message
+               message:
+                 type: string
+                 description: The chat message
+               spacename:
+                 type: string
+                 description: The unique spacename identifier for the message
+
+   operations:
+     sendChatMessage:
+       action: 'send'
+       summary: Send a chat message to a specific room
+       channel:
+         $ref: '#/channels/roomidchat'
+   ```
+3. **Seal the Galactic Document**: Save (`Ctrl + O`, `Enter`) and exit Nano (`Ctrl + X`).
+
+4. **Inform Chan**: Let him know that the foundation for his AsyncAPI document is set.
```

Comment:
```suggestion
3. Save (`Ctrl + O`, `Enter`) and exit Nano (`Ctrl + X`).
```

I'm unsure if we need this step inside Killercoda. It's hard to review this file outside of Killercoda.